checks/evaluation/security_policy.go

```diff
@@ -19,6 +19,105 @@ import (
 	sce "github.com/ossf/scorecard/v4/errors"
 )
 
+func scoreSecurityCriteria(f checker.File,
+	info []checker.SecurityPolicyInformation,
+	dl checker.DetailLogger,
+) int {
+	var urls, emails, discvuls, linkedContentLen, score int
+
+	emails = countSecInfo(info, checker.SecurityPolicyInformationTypeEmail, true)
+	urls = countSecInfo(info, checker.SecurityPolicyInformationTypeLink, true)
+	discvuls = countSecInfo(info, checker.SecurityPolicyInformationTypeText, false)
+
+	for _, i := range findSecInfo(info, checker.SecurityPolicyInformationTypeEmail, true) {
+		linkedContentLen += len(i.InformationValue.Match)
+	}
+	for _, i := range findSecInfo(info, checker.SecurityPolicyInformationTypeLink, true) {
+		linkedContentLen += len(i.InformationValue.Match)
+	}
+
+	msg := checker.LogMessage{
+		Path: f.Path,
+		Type: f.Type,
+		Text: "",
+	}
+
+	// #1: more than one unique (email/http) linked content found: score += 6
+	//     rationale: if more than one link, even stronger for the community
+	if (urls + emails) > 0 {
+		score += 6
+		msg.Text = "Found linked content in security policy"
+		dl.Info(&msg)
+	} else {
+		msg.Text = "no email or URL found in security policy"
+		dl.Warn(&msg)
+	}
+
+	// #2: more bytes than the sum of the length of all the linked content found: score += 3
+	//     rationale: there appears to be information and context around those links
+	//     no credit if there is just a link to a site or an email address (those given above)
+	//     the test here is that each piece of linked content will likely contain a space
+	//     before and after the content (hence the two multiplier)
+	if f.FileSize > 1 && (f.FileSize > uint(linkedContentLen+((urls+emails)*2))) {
+		score += 3
+		msg.Text = "Found text in security policy"
+		dl.Info(&msg)
+	} else {
+		msg.Text = "No text (beyond any linked content) found in security policy"
+		dl.Warn(&msg)
+	}
+
+	// #3: found whole number(s) and or match(es) to "Disclos" and or "Vuln": score += 1
+	//     rationale: works towards the intent of the security policy file
+	//     regarding whom to contact about vuls and disclosures and timing
+	//     e.g., we'll disclose, report a vulnerabily, 30 days, etc.
+	//     looking for at least 2 hits
+	if discvuls > 1 {
```

Comment:
on this final `+1` the regex is looking for digits and portions of the word ‘vuln’ and ‘disclos’ (seen in `checks/raw/security_policy.go`). In this check, there is no test for uniqueness. I did not want to trigger a false positive here where some text have a number in it (like “this is my 1 st security policy”) and nothing else (but I would grant a `+1` with “this is my 1 st and 2 nd security policy”). The “gold” test is looking for hints like “… 90 days for disclosure “.

I felt a single hit was just too easy. Now, the risk is a `SECURITY.md` which has “To report a vulnerability send email to fred@flinstones.com” - this would give you 9 points (the length of free text (+3), the email (+6)), to get the full 10 points you’d improve that to “To report a vulnerability send email to fred@flinstones.com and expect resolution within 90 days” - that’d get you the full 10 points. For me, I was inspired by the Security-Policy check risk statement “possible insecure reporting of vulnerabilities” — so looking for “enough” hints (more than just 1) was my rationale without getting into NLP or following links to other websites to scrape what might be in those files.

Not sure this is resolved here (yet)